WooCommerce/Classes/ViewRelated/Dashboard/Reviews/ReviewsDashboardCardViewModel.swift

```diff
@@ -86,105 +117,162 @@ private extension ReviewsDashboardCardViewModel {
+            self.updateData()
+        }
+        notificationsResultsController.onDidResetContent = { [weak self] in
+            guard let self else { return }
+            self.updateData()
+        }
+
+        do {
+            try notificationsResultsController.performFetch()
+            updateData()
+        } catch {
+            ServiceLocator.crashLogging.logError(error)
+        }
     }
 }
 
 // MARK: - Private helpers
 private extension ReviewsDashboardCardViewModel {
-    @MainActor
-    func loadReviews() async throws {
-        try await withCheckedThrowingContinuation { (continuation: CheckedContinuation<Void, Error>) in
-            stores.dispatch(ProductReviewAction.synchronizeProductReviews(siteID: siteID,
-                                                                          pageNumber: 1,
-                                                                          pageSize: Constants.numberOfItems) { result in
-                switch result {
-                case .success:
-                    // Ignoring the result from remote as we're using storage as the single source of truth
-                    continuation.resume()
-                case .failure(let error):
-                    continuation.resume(throwing: error)
+    /// Populates data from the local storage for the three main data (reviews, products, notifications).
+    ///
+    func updateData() {
+        let localReviews = reviews.prefix(Constants.numberOfItems)
+
+        // We can show partial review content as long as there are reviews found in storage.
+        if localReviews.isEmpty == false {
+            showLoadingAnimation = false
+        }
+
+        data = localReviews.map { review in
+
+            // Depending on the sync progress, `product` and `notification` might still be nil.
+            // This is acceptable and the app is able to display partial review content.
+            let product = reviewProducts.first { $0.productID == review.productID }
+            let notification = notifications.first { notification in
+                if let notificationReviewID = notification.meta.identifier(forKey: .comment) {
+                    return notificationReviewID == review.reviewID
                 }
-            })
+                return false
+            }
+            return ReviewViewModel(
+                showProductTitle: product != nil,
+                review: review,
+                product: product,
+                notification: notification
+            )
         }
     }
 
-    /// Updates data
     @MainActor
-    func updateReviewsResults() async {
-        let reviews = productReviewsResultsController.fetchedObjects.prefix(Constants.numberOfItems)
-        let productIDs = reviews.map { $0.productID }
+    func synchronizeReviews(filter: ReviewsFilter) async throws {
+        try await synchronizeProductReviews(filter: filter)
+
+        let productIDs = reviews.prefix(Constants.numberOfItems).map { $0.productID }
```

Comment:
Mixing storage models when working with network requests can cause flakiness. Here you should work with the items from `synchronizeProductReviews` instead. 

I see that in `synchronizeProductReviews` you ignore the items to use storage as a single source of truth - this only works if you don't have other network requests depending on this request.

Reply:
> I see that in synchronizeProductReviews you ignore the items to use storage as a single source of truth - this only works if you don't have other network requests depending on this request.

Thanks, learned something new about this pattern. Using the first network result for the second request makes sense. Added in https://github.com/woocommerce/woocommerce-ios/pull/12769/commits/3d301e3d2dfeee06c66778271bd5da70168af24e